pkg/types/baremetal/validation/platform.go

```diff
@@ -332,14 +342,24 @@ func ValidatePlatform(p *baremetal.Platform, n *types.Networking, fldPath *field
 func ValidateProvisioning(p *baremetal.Platform, n *types.Networking, fldPath *field.Path) field.ErrorList {
 	allErrs := field.ErrorList{}
 
-	if err := validate.IP(p.BootstrapProvisioningIP); err != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("bootstrapProvisioningIP"), p.BootstrapProvisioningIP, err.Error()))
-	}
-	if err := validateIPNotinMachineCIDR(p.ClusterProvisioningIP, n); err != nil {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("provisioningHostIP"), p.ClusterProvisioningIP, err.Error()))
-	}
+	switch p.ProvisioningNetwork {
+	// If we do not have a provisioning network, provisioning services will be run on the external network. Users must provide IP's and a network interface connected to one of the machine networks.
+	case baremetal.DisabledProvisioningNetwork:
+		// Ensure bootstrapProvisioningIP is in one of the machine networks
+		if err := validateIPinMachineCIDR(p.BootstrapProvisioningIP, n); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("bootstrapProvisioningIP"), p.BootstrapProvisioningIP, fmt.Sprintf("provisioning network is disabled, and %q is not in a machine network", p.BootstrapProvisioningIP)))
+		}
+
+		// Ensure clusterProvisioningIP is in one of the machine networks
+		if err := validateIPinMachineCIDR(p.ClusterProvisioningIP, n); err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("clusterProvisioningIP"), p.ClusterProvisioningIP, fmt.Sprintf("provisioning network is disabled, and %q is not in a machine network", p.ClusterProvisioningIP)))
+		}
 
-	if p.ProvisioningNetworkCIDR != nil {
+		// Make sure the provisioning interface is set.  Very little we can do to validate this as it's not on this machine.
+		if p.ProvisioningNetworkInterface == "" {
```

Comment:
this seems like a validation in both disables and default, so maybe we move it out of the switch.